extensions/wikia/AppPromoLanding/AppPromoLandingController.class.php

```diff
@@ -35,20 +34,25 @@ public function index() {
 		wfProfileIn( __METHOD__ );
 
 		// Since this "Community_App" article won't be found, we need to manually say it's okay so that it's not a 404.
-		$this->response->setCode( self::RESPONSE_OK );
+		$this->response->setCode( static::RESPONSE_OK );
 
 		// Pull in the app-configuration (has data for all apps)
 		$appConfig = [];
 		$memcKey = wfMemcKey( static::$CACHE_KEY, static::$CACHE_KEY_VERSION );
 		$response = $this->wg->memc->get( $memcKey );
 		if ( empty( $response ) ){
-			$req = MWHttpRequest::factory( self::APP_CONFIG_SERVICE_URL, array( 'noProxy' => true ) );
+			$req = MWHttpRequest::factory( static::APP_CONFIG_SERVICE_URL, array( 'noProxy' => true ) );
 			$status = $req->execute();
 			if( $status->isOK() ) {
 				$response = $req->getContent();
 				if ( empty( $response ) ) {
+					// Request failed (app config service failure). Do not cache this response.
+					$this->error(__METHOD__ . ' app config service failure.', [
```

Comment:
Is it really a failure? We already checked if response is ok, maybe we could make this message more descriptive for future debugging?